services/package-json-helpers.js

```diff
@@ -9,13 +20,33 @@ const isDependencyMap = Joi.object()
   )
   .default({})
 
+/**
+ * Joi schema for validating package json object.
+ * Checks if the object has all the dependency types and the dependency types are valid.
+ *
+ * @type {object}
+ */
```

Comment:
Because Joi schemas are so commonly used, we've actually set up jsdoc so that we can use `Joi` as a "type" for schemas

https://github.com/badges/shields/blob/daca76726cc01184b3f851e6436309db90248d3b/.eslintrc.yml#L170-L171

e.g:

https://github.com/badges/shields/blob/daca76726cc01184b3f851e6436309db90248d3b/core/base-service/base-json.js#L29

I guess we've probably merged a few of these PRs describing the schemas as `object` but it lets use `@type {Joi}` for Joi schemas moving forwards.

Reply:
Noted. Joi type does make more sense here. Will remember this moving forward.